projects/hslayers/src/common/layers/interpolated-source.ts

```diff
@@ -0,0 +1,200 @@
+import IDW from 'ol-ext/source/IDW';
+import VectorSource, {LoadingStrategy} from 'ol/source/Vector';
+import {Feature} from 'ol';
+import {GeoJSON} from 'ol/format';
+import {Geometry} from 'ol/geom';
+import {JET_COLOR_MAP} from './jet-color-map.const';
+import {Projection} from 'ol/proj';
+import {Subject} from 'rxjs';
+import {containsExtent, equals} from 'ol/extent';
+
+export const NORMALIZED_WEIGHT_PROPERTY_NAME = 'hs_normalized_IDW_value';
+
+export interface InterpolatedSourceOptions {
+  features?: Feature<Geometry>[];
+  weight?: string;
+  loader?(params: any): Promise<Feature[]>;
+  colorMap?: ((v: number) => number[]) | string;
+  strategy?: LoadingStrategy;
+  maxFeaturesInExtent?: number;
+  maxFeaturesInCache?: number;
+}
+
+export const colorMaps = {'jet': JET_COLOR_MAP};
+
+export class InterpolatedSource extends IDW {
+  featureCache: VectorSource = new VectorSource({});
+  cancelUrlRequest: Subject<void> = new Subject();
+  geoJSONFeatures: string[] = [];
+
+  constructor(private options: InterpolatedSourceOptions) {
+    super({
+      // Source that contains the data
+      source: new VectorSource({
+        strategy:
+          options.strategy != undefined
+            ? options.strategy
+            : (extent: number[], resolution) => {
+                const extentCache = super
+                  .getSource()
+                  .loadedExtentsRtree_.getAll()
+                  .map((item) => item.extent);
+                const toRemove = extentCache.filter(
+                  //Delete cached extents which contain this extent because they have their feature counts limited
+                  (cachedExt) =>
+                    !equals(cachedExt, extent) &&
+                    containsExtent(cachedExt, extent)
+                );
+                for (const extToRemove of toRemove) {
+                  super.getSource().removeLoadedExtent(extToRemove);
+                }
+                return [extent];
+              },
+        loader: async (extent, resolution, projection, success, failure) => {
+          if (options.loader) {
+            this.fillFeatures(
+              await options.loader({
+                extent,
+                resolution,
+                projection,
+                success,
+                failure,
+              }),
+              extent
+            );
+          }
+        },
```

Comment:
Initially I thought that we could provide a default loader like the one that is now used in the test-app example. By just adding `url` as yet another optional parameter to InterpolatedSourceOptions. But I don't know now.. seems to be too intertwined with other Angular-ish code, like proxying, cancelling requests etc.
So I am just leaving it here as a thought 🙃

Reply:
The reason why its like this is the custom format how the features might be returned from the server. Hypothetically it could even be some csv file which doesn't provide feature geometries, but just IDs and values.

Reply:
I see.